src/storage/compression/rle.cpp

```diff
@@ -316,9 +316,41 @@ void RLEScanPartial(ColumnSegment &segment, ColumnScanState &state, idx_t scan_c
 	}
 }
 
+static bool CanEmitConstantVector(idx_t position, idx_t run_length, idx_t scan_count) {
+	D_ASSERT(position < run_length);
+	auto remaining_in_run = run_length - position;
+	// The amount of values left in this run are equal or greater than the amount of values we need to scan
+	return remaining_in_run >= scan_count;
+}
+
+template <class T>
+static void RLEScanConstant(ColumnSegment &segment, RLEScanState<T> &scan_state, idx_t scan_count, Vector &result) {
+	auto data = scan_state.handle.Ptr() + segment.GetBlockOffset();
+	auto data_pointer = (T *)(data + RLEConstants::RLE_HEADER_SIZE);
```

Comment:
I wonder why these casts aren't flagged by clang-tidy. Can we turn these into `reinterpret_cast`?

Reply:
I believe they aren't flagged because it's allowed for templated variables as depending on T, `static_cast` or `reinterpret_cast` might be needed?

I'll change it to use `reinterpret_cast` 👍